jetty-security/src/main/java/org/eclipse/jetty/security/authentication/SpnegoAuthenticator.java

```diff
@@ -106,6 +108,47 @@ public Authentication validateRequest(ServletRequest request, ServletResponse re
         }
     }
 
+    /**
+     * Extracts the auth_scheme from the HTTP Authorization header, {@code Authorization: <auth_scheme> <token>}.
+     *
+     * @param header The HTTP Authorization header or null.
+     * @return The parsed auth scheme from the header, or the empty string.
+     */
+    String getAuthSchemeFromHeader(String header)
+    {
+        // No header provided, return the empty string
+        if (header == null || header.isEmpty())
+        {
+            return "";
+        }
+        // Trim any leading whitespace
+        String trimmed_header = header.trim();
+        // Find the first space, all characters prior should be the auth_scheme
+        int index = trimmed_header.indexOf(' ');
+        if (index > 0) {
+            return trimmed_header.substring(0, index);
+        }
+        // If we don't find a space, this is likely malformed, just return the entire value
+        return trimmed_header;
+    }
+
+    /**
+     * Determines if provided auth scheme text from the Authorization header is case-insensitively
+     * equal to {@code negotiate}.
+     *
+     * @param authScheme The auth scheme component of the Authorization header
+     * @return True if the auth scheme component is case-insensitively equal to {@code negotiate}, False otherwise.
+     */
+    boolean isAuthSchemeNegotiate(String authScheme)
+    {
+        if (authScheme == null || authScheme.isEmpty())
+        {
+            return false;
+        }
+        // Headers should be treated case-insensitively, so we have to jump through some extra hoops.
+        return authScheme.toLowerCase().startsWith(__NEGOTIATE_LC_TEXT);
```

Comment:
This does not do what the javadoc says. It returns true also for `"negotiate-when-hell-freezes-over"`.

I'd suggest just `__NEGOTIATE_LC_TEXT.equalsIgnoreCase(authScheme)`.